### DISCUSSION TITLE
Using Microfrontends with Remix

### DISCUSSION THREAD

**Discussion by ItamarShDev** [remix-run/remix#2648] [Q&A]
Following up on https://github.com/remix-run/remix/discussions/1989;

Suppose I want to create the following microfrontend architecture, how would I go on to achieve it?

Architecture:
- Remix based main host app
- react based frontend only microfrontend apps, with their own remote backends.

In the issue above, the suggestion is for a solution where all apps are Remix based, but in my case, besides the host, all apps are Python backend with a standalone React only frontend.

Is there any build in approach to this?

Thank you!

**Comment by sergiodxa** [comment]
If each app owns different URLs, you could use a reverse proxy before your Remix and plain React app, and make certain routes go to Remix and other go to plain React.

**Reply by ItamarShDev** [reply]
Thank you for the answer, i would be happy to followup a bit on that:
Should i use Module Federation to render the micro-frontends inside the remix app?

If not, what is the right approach in achieving it?

**Reply by sergiodxa** [reply]
Module Federation is only supported on webpack, I don’t know how you could use it inside an app not using webpack for the build like Remix

**Reply by ItamarShDev** [reply]
Thanks, is there any supported method to teach this goal?
I want to done a way to get the apps at rib time

**Comment by digitalsadhu** [comment]
Unsure how helpful this is but you could use https://podium-lib.io (Full disclosure, I'm a maintainer). It's easy in Next and remix apps to use the Podium client to fetch content from page fragment frontend servers and stitch together your page.

**Reply by ankarhem** [reply]
Hey, full disclosure I just skimmed the docs but couldn't find anything about how shared dependencies are handled?

**Reply by digitalsadhu** [reply]
When you say shared dependencies, you’re referring to client side dependencies such as react etc? We wrote another project to allow us to align shared deps across microfrontend apps. https://eik.dev/ Unfortunately since Remix doesn’t give access to the build pipeline to add an Eik plugin, we have to do some very hacky things to make this approach work.

**Reply by ankarhem** [reply]
Yes, precisely. Currently evaluating different paths forward with MFE. I’ll check this out, thanks.

**Reply by cliffordfajardo** [reply]
Don't have a solution for your original microfrontend question, but I wanted to share that in  about a month @adrienbaron will have a tech talk at Remix Conf Europe about Remix & Microfrontends strategies
- https://remixconf.eu/

**Reply by adrienbaron** [reply]
Thanks for shoutout @cliffordfajardo ! I'll be talking mostly about horizontal slicing using the lib I built while at Cazoo: https://tiny-frontend.github.io/, as well as the experiment I did around vertical slicing with Remix (that this discussion is a follow up of)
